### PR TITLE
Merge selectors with ListOptions.

### PR DESCRIPTION
**Pull request by wojtek-t**
This is followup/cleanup from previous PRs.

@smarterclayton @lavalamp

### CODE REVIEW

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

File: pkg/registry/generic/etcd/etcd.go [hunk 1]
```diff
@@ -159,12 +159,20 @@ func (e *Etcd) NewList() runtime.Object {
...
+	if options != nil && options.FieldSelector != nil {
+		field = options.FieldSelector
+	}
+	return e.listPredicate(ctx, e.PredicateFunc(label, field), options)
 }
 
-// ListPredicate returns a list of all the items matching m.
-func (e *Etcd) ListPredicate(ctx api.Context, m generic.Matcher, options *api.ListOptions) (runtime.Object, error) {
+// listPredicate returns a list of all the items matching m.
+func (e *Etcd) listPredicate(ctx api.Context, m generic.Matcher, options *api.ListOptions) (runtime.Object, error) {
```
Comment (smarterclayton) [inline pkg/registry/generic/etcd/etcd.go]: We use ListPredicate in various contexts - it needs to remain public.  We can make it private in a separate followup.

**Comment by smarterclayton** [comment]
One comment

File: pkg/registry/generic/etcd/etcd.go (hunk 1 above)
Comment (wojtek-t) [inline pkg/registry/generic/etcd/etcd.go]: Done.

**Comment by wojtek-t** [comment]
@smarterclayton - thanks for review. PTAL

**Comment by smarterclayton** [comment]
LGTM

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

**Pull request merged**